packages/mdl-drawer/temporary/foundation.js

```diff
@@ -0,0 +1,253 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MDLFoundation} from 'mdl-base';
+import {cssClasses, strings, numbers} from './constants';
+
+export default class MDLTemporaryDrawerFoundation extends MDLFoundation {
+  static get cssClasses() {
+    return cssClasses;
+  }
+
+  static get strings() {
+    return strings;
+  }
+
+  static get numbers() {
+    return numbers;
+  }
+
+  static get defaultAdapter() {
+    return {
+      addClass: (/* className: string */) => {},
+      removeClass: (/* className: string */) => {},
+      hasClass: (/* className: string */) => {},
+      hasNecessaryDOM: () => /* boolean */ {},
+      registerComponentInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      deregisterComponentInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      registerDrawerInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      deregisterDrawerInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      registerTransitionEndHandler: (/* handler: EventListener */) => {},
+      deregisterTransitionEndHandler: (/* handler: EventListener */) => {},
+      updateDrawerPosition: (/* value: number | null */) => {},
+      updateComponentBackground: (/* value: string */) => {},
+      getFocusableElements: () => /* array<Element> */ {},
+      saveElementTabState: (/* el: Element */) => {},
+      restoreElementTabState: (/* el: Element */) => {},
+      makeElementUntabbable: (/* el: Element */) => {},
+      isRightSide: () => /* boolean */ {}
+    };
+  }
+
+  constructor(adapter) {
+    super(Object.assign(MDLTemporaryDrawerFoundation.defaultAdapter, adapter));
+
+    this.transitionEndHandler_ = () => {
+      this.adapter_.removeClass(MDLTemporaryDrawerFoundation.cssClasses.ANIMATING);
+      this.adapter_.deregisterTransitionEndHandler(this.transitionEndHandler_);
+    };
+
+    this.inert_ = false;
+
+    this.componentClickHandler_ = () => this.close();
+    this.drawerClickHandler_ = evt => evt.stopPropagation();
+    this.componentTouchStartHandler_ = evt => this.handleTouchStart_(evt);
+    this.componentTouchMoveHandler_ = evt => this.handleTouchMove_(evt);
+    this.componentTouchEndHandler_ = evt => this.handleTouchEnd_(evt);
+  }
+
+  init() {
+    const {ROOT, OPEN} = MDLTemporaryDrawerFoundation.cssClasses;
+
+    if (!this.adapter_.hasClass(ROOT)) {
```

Comment:
Pretty much. I think it's a good pattern to make sure that a component has everything it needs to function correctly. That means checking for the base class and checking for the required DOM (the bits of it that really need to be there). That way we can preempt problems and provide better error messages.

I'm happy moving these checks to `mdl-base`, though. I can see checking for base class and required DOM (we could have a `return true` `hasNecessaryDOM` method in `mdl-base`) as part of the init work on `mdl-base`.